### PR TITLE
feat: add types

### PR DESCRIPTION
**Pull request by achingbrain**
Adds ts types, fixes all tsc errors, updates all deps.

Removes ipfs/ipfsd-ctl/ipfs-http-client dev deps as we only spin up
a node to get an IPFS id which we can just hard code for testing purposes.

Replaces protons with protobufjs as elsewhere in the stack.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -12,38 +12,35 @@ const uint8ArrayToString = require('uint8arrays/to-string')
...
 
-const ipnsEntryProto = require('./pb/ipns.proto')
+const {
+  IpnsEntry: ipnsEntryProto
+} = require('./pb/ipns.js')
 const { parseRFC3339 } = require('./utils')
 const ERRORS = require('./errors')
 
-const ID_MULTIHASH_CODE = multihash.names.id
+const ID_MULTIHASH_CODE = multihash.names.identity
```
Comment (achingbrain) [inline src/index.js:25]: 😱

File: src/index.js [hunk 2]
```diff
@@ -116,7 +118,7 @@ const validate = async (publicKey, entry) => {
       throw errCode(new Error('unrecognized validity format (not an rfc3339 format)'), ERRORS.ERR_UNRECOGNIZED_FORMAT)
     }
 
-    if (validityDate < Date.now()) {
+    if (validityDate.getTime() < Date.now()) {
```
Comment (achingbrain) [inline src/index.js:121]: 😱

File: package.json [hunk 3]
```diff
@@ -4,8 +4,13 @@
...
   "leadMaintainer": "Vasco Santos <vasco.santos@moxy.studio>",
   "main": "src/index.js",
+  "types": "dist/src/index.d.ts",
   "scripts": {
-    "build": "aegir build",
+    "prepare": "run-s prepare:*",
+    "prepare:proto": "pbjs -t static-module -w commonjs --force-number --no-verify --no-delimited --no-create --no-beautify --no-defaults --lint eslint-disable -o src/pb/ipns.js src/pb/ipns.proto",
+    "prepare:proto-types": "pbts -o src/pb/ipns.d.ts src/pb/ipns.js",
+    "prepare:types": "aegir build --no-bundle",
+    "prepublishOnly": "aegir build",
```
Comment (hugomrdias) [inline package.json:13]: release cmd already runs aegir build

**Review by hugomrdias — APPROVED** [review]

File: package.json (hunk 3 above)
Comment (achingbrain) [reply package.json:13]: Magic

File: src/index.js (hunk 1 above)
Comment (vasco-santos) [reply src/index.js:25]: This used to be `.id` but apparently not flagged as a breaking change: https://github.com/multiformats/js-multihash/pull/66

Also, somehow we don't cover this in the tests 😱

**Review by vasco-santos — APPROVED** [review]
LGTM

**Comment by vasco-santos** [comment]
Shipped `0.9.0` to guarantee nothing breaks upstream with the types

**Pull request merged**